### ISSUE TITLE
Cannot return a struct when it has an array in it

### ISSUE THREAD

**Issue by michwill** [vyperlang/vyper#1839]
### Version Information

* vyper Version (output of `vyper --version`): 0.1.0b16
* OS: linux

Example causing an issue:
```python
Z256: constant(uint256) = 0
ZEROS: constant(uint256[3]) = [Z256, Z256, Z256]

struct A:
    single: uint256
    arr: uint256[3]

@constant
@private
def f() -> A:
    x: A = A({single: 0, arr: ZEROS})
    return x
```

Result:
```
Error compiling: structbug.vy
vyper.exceptions.TypeMismatchException: line 12:4 Invalid type for array index: None
```

**Comment by michwill** [comment]
In addition, similar error happens when it's:
```python
def f() -> (uint256, uint256[3]):
    ...
    return (some_val, some_arr)
```

**Comment by fubuloubu** [comment]
Related: #1409

**Comment by fubuloubu** [comment]
Compiles in master! Fixed as #1838
